decorator.rb

```diff
@@ -0,0 +1,15 @@
+# decorator.rb
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_reader :nameable
+
+  def initialize(nameable)
+    super()
```

Comment:
- [ ] Hello! I've noticed that you're calling super within the initialize method of the Decorator class. While Decorator does inherit from the Nameable class, it's important to note that in this specific scenario, using super isn't necessary. This is because the Nameable class doesn't pass down any attributes or parameters to its subclasses, aside from the correct_name method.

👍 In other words, since there are no additional attributes or values being inherited from Nameable, there's no need to use super in the initialize method of Decorator. This simplifies your code and eliminates any potential confusion. 🚀🔧

Reply:
Hey @zeff96 😊 Just a quick heads-up – when I remove the super call in the initialize method of the Decorator class, it might cause a bit of a fuss with linter errors as it is generally recommended for proper initialization of the parent class. It ensures that any required initialization in the parent class is executed, even when there are no additional attributes being inherited. This practice improves code maintainability and helps prevent potential issues in the future

Reply:
@Rafiduzzaman, I get that, in this scenario though there is no initialization in the parent class, but sure if in the future you see a possibility of initializing the superclass then that is ok